tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import re
+
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required, ignore_leftovers, ipi_deployment_required)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import ocp
+from ocs_ci.ocs import node
+from tests.sanity_helpers import Sanity
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@ignore_leftovers
+@aws_platform_required
+@ipi_deployment_required
+class Testknip894(ManageTest):
+    """
+    Knip-678 Node Replacement
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_nodereplacement_proactive(self, pvc_factory, pod_factory):
+        """
+        Knip-894 Node Replacement proactive
+        """
+        self.sanity_helpers.create_resources(pvc_factory, pod_factory)
+        ocs_worker_node = node.get_ocs_worker_nodes()
```

Comment:
this will be done by ocp while deleting machine

Reply:
we have documented to do unschedule and drain before draining. So, it would be good to follow the same.
If this is not required and we have some confirmation from ocp team, we will have to update the docs then.

Reply:
Will update in next commit

Reply:
as per logs of machine-api-controllers -c machine-controllers  drain of pod is handled by machine deleted

```
Reconciling Machine "pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj"
I0121 13:45:50.246830       1 controller.go:304] Machine "pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj" in namespace "openshift-machine-api" doesn't specify "cluster.k8s.io/cluster-name" label, assuming nil cluster
I0121 13:45:50.246847       1 controller.go:192] Reconciling machine "pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj" triggers delete
I0121 13:45:50.350375       1 info.go:16] ignoring DaemonSet-managed pods: tuned-4rfqj, dns-default-c68lc, node-ca-65j2g, machine-config-daemon-4fp5g, node-exporter-gr267, multus-7tnzn, ovs-d9l26, sdn-h4qtg, csi-cephfsplugin-zhdqq, csi-rbdplugin-92ftf; deleting pods with local storage: rook-ceph-osd-0-5ff69458f9-c9kb8
I0121 13:45:57.383523       1 info.go:20] pod "ocs-operator-6cb6f48dc6-d9526" removed (evicted)
I0121 13:45:57.387873       1 info.go:20] pod "rook-ceph-osd-0-5ff69458f9-c9kb8" removed (evicted)
I0121 13:45:57.387911       1 info.go:20] drained node "ip-10-0-147-239.us-east-2.compute.internal"
I0121 13:45:57.387928       1 controller.go:296] drain successful for machine "pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj"
I0121 13:45:57.387972       1 actuator.go:333] pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj: deleting machine
I0121 13:45:57.474289       1 utils.go:152] Cleaning up extraneous instance for machine: i-096c75bd5370b4f48, state: running, launchTime: 2020-01-21 07:04:56 +0000 UTC
I0121 13:45:57.474321       1 utils.go:156] Terminating i-096c75bd5370b4f48 instance
I0121 13:45:57.605600       1 controller.go:219] Deleting node "ip-10-0-147-239.us-east-2.compute.internal" for machine "pratik-42-ocs-35-nfff7-worker-us-east-2b-gjvsj"
```